spec/unit/puppet/provider/cs_colocation_crm_spec.rb

```diff
@@ -0,0 +1,140 @@
+require 'spec_helper'
+
+describe Puppet::Type.type(:cs_colocation).provider(:crm) do
+  before do
+    described_class.stubs(:command).with(:crm).returns 'crm'
+  end
+
+  context 'when getting instances' do
+    let :instances do
+
+      test_cib = <<-EOS
+        <cib>
+        <configuration>
+          <constraints>
+            <rsc_colocation id="first_with_second" rsc="first" score="INFINITY" with-rsc="second"/>
+          </constraints>
+        </configuration>
+        </cib>
+      EOS
+
+      described_class.expects(:block_until_ready).returns(nil)
+      if Puppet::PUPPETVERSION.to_f < 3.4
+        Puppet::Util::SUIDManager.expects(:run_and_capture).with(['crm', 'configure', 'show', 'xml']).at_least_once.returns([test_cib, 0])
+      else
+        Puppet::Util::Execution.expects(:execute).with(['crm', 'configure', 'show', 'xml']).at_least_once.returns(
+          Puppet::Util::Execution::ProcessOutput.new(test_cib, 0)
+        )
+      end
+      instances = described_class.instances
+    end
+
+    it 'should have an instance for each <colocation>' do
+      expect(instances.count).to eq(1)
+    end
+
+    describe 'each instance' do
+      let :instance do
+        instances.first
+      end
+
+      it "is a kind of #{described_class.name}" do
+        expect(instance).to be_a_kind_of(described_class)
+      end
+
+      it "is named by the <primitive>'s id attribute" do
+        expect(instance.name).to eq("first_with_second")
+      end
+
+      it "should have attributes" do
+        expect(instance.primitives).to eq(['second', 'first'])
+        expect(instance.score).to eq('INFINITY')
+      end
+    end
+  end
+
+  context 'when flushing' do
+    def expect_update(pattern)
+      instance.expects(:crm).with { |*args|
+        if args.slice(0..2) == ['configure', 'load', 'update']
+          expect(File.read(args[3])).to match(pattern)
+          true
```

Comment:
the return value of `expect_update` is never checked. you can leave it out then?